src/validate.ts

```diff
@@ -111,3 +112,14 @@ export function generateDisconnectRequest(){
         }]
     }
 }
+
+type ApiResponse = {[key: string]: any}
+const sync_response_schema = require('../intents/sync.response.schema.json');
+export function validate(apiResponse: ApiResponse, responseType: string){
```

Comment:
You can make the `responseType` a specific type or enum of strings. ie. `responseType: 'sync' | 'query' ...` to more precisely target the possible values.

Alternatively you can have an explicit `validateSyncResponse` function.